packages/core/src/validate/run.ts

```diff
@@ -629,6 +661,129 @@ function* getInternalFunctionStorageErrors(
+          ) {
+            const referencedFn = fnCall.expression.referencedDeclaration;
+
+            // If this is a call to a parent initializer, then:
+            // - Check for duplicates
+            // - Check if the parent initializer is called in the correct order
+            for (const [baseName, initializers] of baseContractsInitializersMap) {
+              const foundParentInitializer = initializers.find(init => init.id === referencedFn);
+              if (referencedFn && foundParentInitializer) {
+                const duplicate = calledInitializerIds.includes(referencedFn);
+                if (
+                  duplicate &&
+                  !skipCheck('duplicate-initializer-call', contractDef) &&
+                  !skipCheck('duplicate-initializer-call', contractInitializer) &&
+                  !skipCheck('duplicate-initializer-call', stmt)
+                ) {
+                  yield {
+                    kind: 'duplicate-initializer-call',
+                    src: decodeSrc(fnCall),
+                    parentInitializer: foundParentInitializer.name,
+                    parentContract: baseName,
+                  };
+                }
+                calledInitializerIds.push(referencedFn);
+
+                const index = uninitializedBaseContracts.indexOf(baseName);
+                if (
+                  !duplicate &&
+                  index !== 0 &&
+                  !skipCheck('incorrect-initializer-order', contractDef) &&
+                  !skipCheck('incorrect-initializer-order', contractInitializer)
+                ) {
+                  yield {
+                    kind: 'incorrect-initializer-order',
+                    src: decodeSrc(fnCall),
+                    expectedLinearization: baseContractsWithInitializers,
+                  };
+                }
+                if (index !== -1) {
+                  uninitializedBaseContracts.splice(index, 1);
+                }
+              }
+            }
+          }
+        }
+
+        // If there are any base contracts that were not initialized, report an error
+        if (
+          uninitializedBaseContracts.length > 0 &&
+          !skipCheck('missing-initializer-call', contractDef) &&
+          !skipCheck('missing-initializer-call', contractInitializer)
+        ) {
+          yield {
+            kind: 'missing-initializer-call',
+            src: decodeSrc(contractInitializer),
+            parentContracts: uninitializedBaseContracts,
+          };
+        }
+      }
+    }
+  }
+}
+
+function getPossibleInitializers(contractDef: ContractDefinition, isParentContract: boolean) {
+  const fns = [...findAll('FunctionDefinition', contractDef)];
+  return fns.filter(
+    fnDef =>
+      (fnDef.modifiers.some(modifier =>
+        ['initializer', 'reinitializer', 'onlyInitializing'].includes(modifier.modifierName.name),
+      ) ||
+        ['initialize', 'initializer', 'reinitialize', 'reinitializer'].includes(fnDef.name)) &&
```

Comment:
For another PR, I think it may be useful if we annotate initializers with `@custom:initializer` or similar.
The idea is that there are contracts with initializers that don't follow the `initialize` or `initializer` convention in its name.

A concrete example is any of the Signers we created for ERC4337 accounts:

```solidity
abstract contract SignerRSA is AbstractSigner {
    error SignerRSAUninitializedSigner(bytes e, bytes n);

    bytes private _e;
    bytes private _n;

    /// @custom:initializer true
    function _initializeSigner(bytes memory e, bytes memory n) internal {
        if (_e.length != 0 || _n.length != 0) revert SignerRSAUninitializedSigner(e, n);
        _e = e;
        _n = n;
    }

...
}
```